mod/wikirate_source/set/type/source/source.rb

```diff
@@ -175,66 +178,68 @@ def parse_source_page url
 end
 
 format :html do
-
   def edit_slot args
     # see claim.rb for explanation of core_edit
-    super args.merge(:core_edit=>true)
+    super args.merge(core_edit: true)
   end
 
   view :metric_import_link do |args|
-    file_card = Card[card.name+"+File"]
-    if file_card and mime_type = file_card.file.content_type and ( mime_type == "text/csv" || mime_type == "text/comma-separated-values" )
-      card_link file_card, {:text=>"Import to metric values",:path_opts=>{:view=>:import}}
-    else
-      ""
-    end
+    file_card = Card[card.name+'+File']
+    return '' unless has_import_mime_type?(file_card)
+    card_link file_card, text: "Import to metric values",
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.